safe-global/safe-wallet-web
# feat: redesign approval editor

## What it solves

Resolves #3607 

## TODO
- [x] Maintain order of approvals as they are in the transaction or fix update function

## How this PR fixes it
- Changes layout / design of approval editor
- Groups approvals by spender address
- Fetches and displays contract name for spender address

## How to test it
- Use any app that requires approvals OR transaction builder to craft specific approval payloads
- Observe the approval editor


## Screenshots
<img width="449" alt="Screenshot 2024-04-25 at 13 46 12" src="https://github.com/safe-global/safe-wallet-web/assets/2670790/3f9ec297-ab11-4f79-82d0-632c851951a7">
<img width="449" alt="Screenshot 2024-04-25 at 13 46 19" src="https://github.com/safe-global/safe-wallet-web/assets/2670790/49b24e11-c3f9-43a7-a2c1-e97cb8b94284">
<img width="449" alt="Screenshot 2024-04-25 at 13 46 26" src="https://github.com/safe-global/safe-wallet-web/assets/2670790/7723eccf-5b06-47cc-b44f-95732f76ed23">
<img width="587" alt="Screenshot 2024-04-25 at 13 28 17" src="https://github.com/safe-global/safe-wallet-web/assets/2670790/6b62c02c-1c5f-4c98-a6cb-9811ba82f5d8">


## Checklist
* [] I've written a unit/e2e test for it (if applicable) 🧑‍💻
